pkg/plugincmdtree/plugins_cache_test.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package plugincmdtree
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/cli"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+)
+
+const samplePluginToGenerateAliasWithHelpCommand string = `#!/bin/bash
+
+# Dummy Tanzu CLI 'Plugin' to return the help output so that parser can extract aliases
+
+if [ "$1" = "-h" ]; then
+  echo "fake plugin help"
+  echo "Aliases:"
+  echo "  sp"
+elif [ "$1" = "sub-command1" ] && [ "$2" = "-h"  ]; then
+    echo "fake sub-command1 with aliases"
+    echo "Aliases:"
+    echo "  sub-command1, sc1"
+elif [ "$1" = "sub-command2" ] && [ "$2" = "-h"  ]; then
+    echo "fake sub-command2 without aliases"
+elif [ "$1" = "sub-command1" ] && [ "$2" = "sub-command1-sub" ] && [ "$3" = "-h" ]; then
+  echo "fake sub-command1-sub with aliases"
+  echo "Aliases:"
+  echo "  sub-command1-sub, sc1s, clusters"
+else
+  echo "Invalid command."
+fi
+`
+
+const expectedPluginCmdTree string = `
+commandTree:
+  ? pluginName: sample-plugin
+    target: kubernetes
+    version: 1.0.0
+  : subcommands:
+      sub-command1:
```

Comment:
nit: sub commands prefixed with 'sub' or 'sub-command' mingled with yaml nodes whose names are 'subcommands' makes  for a rather confusing read :-). I would have picked other set of names with some well know order (e.g. foo, bar, baz, foo1, foo2, ..)

Also I am not sure what happened to sub-command1-sub and the aliases in the expectedPluginCmdTree.

Reply:
ACK. Updated the command name to foo1,bar1 and sub-subcommand to follow the order foo2(sub-coomand of foo1). 
Also fixed the issue with `sub-command1-sub` aliases (doc for the same was not pre-generated in the test hence it was missed)